sdk/python/kfp_tekton/compiler/compiler.py

```diff
@@ -320,6 +320,37 @@ def _create_pipeline_workflow(self, args, pipeline, op_transformers=None, pipeli
       if op.timeout:
         task['timeout'] = '%ds' % op.timeout
 
+    # handle resourceOp cases in pipeline
+    for task in task_refs:
+      op = pipeline.ops.get(task['name'])
+      if isinstance(op, dsl.ResourceOp):
+        action = op.resource.get('action')
+        merge_strategy = op.resource.get('merge_strategy')
+        success_condition = op.resource.get('success_condition')
+        failure_condition = op.resource.get('failure_condition')
+        task['params'] = [tp for tp in task.get('params', []) if tp.get('name') != "image"]
+        if not merge_strategy:
+          task['params'] = [tp for tp in task.get('params', []) if tp.get('name') != 'merge-strategy']
+        if not success_condition:
+          task['params'] = [tp for tp in task.get('params', []) if tp.get('name') != 'success-condition']
+        if not failure_condition:
+          task['params'] = [tp for tp in task.get('params', []) if tp.get('name') != "failure-condition"]
+        for tp in task.get('params', []):
+          if tp.get('name') == "action" and action:
+            tp['value'] = action
+          if tp.get('name') == "merge-strategy" and merge_strategy:
+            tp['value'] = merge_strategy
+          if tp.get('name') == "success-condition" and success_condition:
+            tp['value'] = success_condition
+          if tp.get('name') == "failure-condition" and failure_condition:
+            tp['value'] = failure_condition
+          if tp.get('name') == "manifest":
+            manifest = yaml.dump(convert_k8s_obj_to_json(op.k8s_resource), default_flow_style=False)
+            tp['value'] = manifest.replace('{{workflow.name}}', pipeline.name)
```

Comment:
At some point we probably need to replace this because `{{workflow.name}}` is unique for every workflow run where `pipeline.name` is static. Can we add a todo comment here? Since Tekton already have pipelinerun name in the task's label, if you somehow can pass pipelinerun name using `fieldRef` then it can also work as well.

Reply:
Just removed it, since resourceOp don't have {{workflow.name}} defined, it was appeared in VolumeOp.

Reply:
Ok then I should put it as a todo for the VolumeOp and volumesnapshotop.